src/containers/VisualElement/VisualElement.tsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2017-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React, { useMemo } from 'react';
+import VisualElementEditor from '../../components/SlateEditor/VisualElementEditor';
+import visualElementPlugin from '../../components/SlateEditor/plugins/visualElement';
+import visualElementPickerPlugin from '../../components/SlateEditor/plugins/visualElementPicker';
+
+interface Props {
+  onChange: Function;
+  changeVisualElement: Function;
+  name: string;
+  types: string[];
+  language: string;
+  value: any;
+}
+
+const createPlugins = (
+  empty: boolean,
+  types: string[],
+  changeVisualElement: Function,
+  language: string,
+) => {
+  return [
+    visualElementPickerPlugin({
+      types,
+      changeVisualElement,
+      empty,
+      language,
+    }),
+    visualElementPlugin({
+      language,
+    }),
+  ];
+};
+
+const VisualElement = ({
+  onChange,
+  changeVisualElement,
+  name,
+  types,
+  language,
+  value,
+}: Props) => {
+  const plugins = useMemo(() => {
+    return createPlugins(!value, types, changeVisualElement, language);
```

Comment:
Ved opprettelse av nytt emne er value et tomt objekt, mens om du fjerner visuelt element fra en emneartikkel er det undefined. Denne varianten viser plussen i begge tilfella. Ikkje optimalt, men noko som funker inntil videre.
```suggestion
    return createPlugins(
      !value || Object.keys(value).length === 0,
      types,
      changeVisualElement,
      language,
    );
```